From 167e22ed2aee70811fe520fdf996b11156450a07 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:12 -0400
Subject: [serial] tty/hvc_opal: powerpc - Make OPAL HVC device tree accesses endian safe

Message-id: <1410545655-205645-84-git-send-email-dzickus@redhat.com>
Patchwork-id: 94128
O-Subject: [RHEL7 PATCH 083/626] tty/hvc_opal: powerpc: Make OPAL HVC device tree accesses endian safe
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5c94913c5f9834b0d35e20359db457783962bed5
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Mon Sep 23 12:05:08 2013 +1000

    tty/hvc_opal: powerpc: Make OPAL HVC device tree accesses endian safe

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/tty/hvc/hvc_opal.c b/drivers/tty/hvc/hvc_opal.c
index 27e8719..73f9f53 100644
--- a/drivers/tty/hvc/hvc_opal.c
+++ b/drivers/tty/hvc/hvc_opal.c
@@ -349,7 +349,7 @@ static void udbg_init_opal_common(void)
 void __init hvc_opal_init_early(void)
 {
  struct device_node *stdout_node = NULL;
- const u32 *termno;
+ const __be32 *termno;
  const char *name = NULL;
  const struct hv_ops *ops;
  u32 index;
@@ -391,7 +391,7 @@ void __init hvc_opal_init_early(void)
  if (!stdout_node)
   return;
  termno = of_get_property(stdout_node, "reg", NULL);
- index = termno ? *termno : 0;
+ index = termno ? be32_to_cpup(termno) : 0;
  if (index >= MAX_NR_HVC_CONSOLES)
   return;
  hvc_opal_privs[index] = &hvc_opal_boot_priv;
-- 
1.7.1